src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/Statics.cs

```diff
@@ -503,7 +503,7 @@ public static Delegate CreateDelegate(Type delegateType, MethodInfo methodInfo)
 
             EventDataAttribute? eventAttrib = Statics.GetCustomAttribute<EventDataAttribute>(dataType);
             if (eventAttrib != null ||
-                Statics.GetCustomAttribute<CompilerGeneratedAttribute>(dataType) != null ||
+                IsCompilerGenerated(dataType) ||
```

Comment:
If I am reading this correctly, turning off Debugger support would lead to silent behavior changes in Tracing. It is unpredictable experience.

Reply:
This change isn't critical since EventSource is disabled by default in Blazor. I included it for completeness, but I can undo it. The resulting behavior would be if someone trimmed their app with `Debugger.IsSupported=false` and `EventSource.IsSupported=true`, they would still get `CompilerGeneratedAttributes` in their binaries (after https://github.com/mono/linker/issues/1378 is fixed). So the size would grow slightly, but `EventSource` would work the exact same.

As an aside, I'm a little bit confused by the existing code here. When would someone pass a compiler generated type into `EventSource.Write`? Is this how we check for anonymous types?

cc @brianrob @noahfalk

Reply:
If we don't need I agree we should leave it as-is for now.

Reply:
> Is this how we check for anonymous types?

Yes, we document that in the EventSource.Write docs: "The event data. This type must be an anonymous type or marked with the EventDataAttribute attribute." and this check is trying to do that. There isn't a better way to check for anonymous types that I know of.